core/src/main/java/com/google/zxing/qrcode/detector/FinderPatternFinder.java

```diff
@@ -604,85 +613,52 @@ private boolean haveMultiplyConfirmedCenters() {
       throw NotFoundException.getNotFoundInstance();
     }
 
-    // Filter outlier possibilities whose module size is too different
-    if (startSize > 3) {
-      // But we can only afford to do so if we have at least 4 possibilities to choose from
-      double totalModuleSize = 0.0;
-      double square = 0.0;
-      for (FinderPattern center : possibleCenters) {
-        float size = center.getEstimatedModuleSize();
-        totalModuleSize += size;
-        square += (double) size * size;
-      }
-      double average = totalModuleSize / startSize;
-      float stdDev = (float) Math.sqrt(square / startSize - average * average);
+    Collections.sort(possibleCenters, new EstimatedModuleComparator());
 
-      Collections.sort(possibleCenters, new FurthestFromAverageComparator((float) average));
+    double distotion = Double.MAX_VALUE;
+    double[] squares = new double[3];
+    FinderPattern[] bestPatterns = new FinderPattern[3];
 
-      float limit = Math.max(0.2f * (float) average, stdDev);
+    for (int i = 0; i < possibleCenters.size() - 2; i++) {
+      for (int j = i + 1; j < possibleCenters.size() - 1; j++) {
+        for (int k = j + 1; k < possibleCenters.size(); k++) {
+          float minModuleSize = possibleCenters.get(i).getEstimatedModuleSize();
+          float maxModuleSize = possibleCenters.get(k).getEstimatedModuleSize();
+          if (maxModuleSize > minModuleSize * 1.4f) {
```

Comment:
The limit of the module size in original code was average module size +/- 20%.
So I make a new limit +40% of smallest module size.